src/data/fintype/basic.lean

```diff
@@ -624,6 +624,12 @@ by simp [to_finset]
 @[simp] theorem mem_to_finset_val {s : set α} [fintype s] {a : α} : a ∈ s.to_finset.1 ↔ a ∈ s :=
 mem_to_finset
 
+/-- Many `fintype` instances for sets are defined using an extensionally equal `finset`.
+Rewriting `s.to_finset` with `set.to_finset_of_finset` replaces the term with such a `finset`. -/
+theorem to_finset_of_finset {p : set α} (s : finset α) (H : ∀ x, x ∈ s ↔ x ∈ p) :
```

Comment:
Can this be a reasonable `simp` lemma?

Reply:
I decided not to because it would conflict with the simp lemmas for `(s.to_finset : set X) = s`, which seemed to be better to preserve.

This seems to be like a saddle point in the simp set: do we prefer going into the language of `set` or the language of `finset`? Sometimes it seems like we should be careful with having additional simp attributes that can guide simp toward a particular sub-language.